contracts/zero-ex/contracts/src/transformers/LibERC20Transformer.sol

```diff
@@ -36,19 +36,21 @@ library LibERC20Transformer {
     ///      This is just `keccak256('TRANSFORMER_SUCCESS')`.
     bytes4 constant internal TRANSFORMER_SUCCESS = 0x13c9929e;
 
-    /// @dev Transfer ERC20 tokens and ETH.
+    /// @dev Transfer ERC20 tokens and ETH. For ETH transfer, it's not safe from re-entrancy attacks and the
+    // caller is responsible for gurading against a potential re-entrancy attack.
     /// @param token An ERC20 or the ETH pseudo-token address (`ETH_TOKEN_ADDRESS`).
     /// @param to The recipient.
     /// @param amount The transfer amount.
-    function transformerTransfer(
+    function unsafeTransformerTransfer(
         IERC20TokenV06 token,
         address payable to,
         uint256 amount
     )
         internal
     {
         if (isTokenETH(token)) {
-            to.transfer(amount);
+            (bool sent,) = to.call{value: amount}("");
```

Comment:
So is `to.call{}()`  not limited to fixed 2300 gas? Do we want/need to specify max gas here in `call{gas: X}("")`? 
Why did you change the func name to unsafe...? as in what are the safety differences between transfer and call in this context?

Reply:
`call` isn't limited to 2300 -- if we don't specify `gas` that would be forward all gas (that is what we're doing here). we could potentially specify a gas limit, but I don't think there is much benefit of doing that (would like to hear what @dekz  thinks).

> unsafe

if not safe from [re-entrancy](https://quantstamp.com/blog/what-is-a-re-entrancy-attack) attack. Since it's a common library that can be used by other contracts in the future, `unsafe` prefix is there to remind the caller to be extra cautious (suggested by @dekz).

Reply:
Best summary I found for transfer/send/call:

![image](https://user-images.githubusercontent.com/6490062/192919650-7ce13c7b-a262-4040-b48f-907ab74ff2ea.png)